From 857c2738e8e8f35e7135332747a37df6bacc618d Mon Sep 17 00:00:00 2001
From: Ruchi Kandoi <kandoiruchi@google.com>
Date: Mon, 27 Nov 2017 13:17:43 -0800
Subject: [PATCH 3/4] Correct the parameter length for core_initialized()

Ib8ebcfb5f3998bba025e9559ab52e1f9eacc4df6 introduced a bug where the
length of the argument for core_initialized() was incorrect, causing
data to be truncated when sent over the HAL interface.

Test: NFC enable/disable
Bug: 69438157

Change-Id: Iaa5e66f2191da20340c31e0c6c2725b60dab245b
---
 src/nfc/nfc/nfc_ncif.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/nfc/nfc/nfc_ncif.cc b/src/nfc/nfc/nfc_ncif.cc
index da691a4..b702611 100644
--- a/src/nfc/nfc/nfc_ncif.cc
+++ b/src/nfc/nfc/nfc_ncif.cc
@@ -1373,7 +1373,7 @@ void nfc_ncif_proc_init_rsp(NFC_HDR* p_msg) {
       nfc_set_state(NFC_STATE_W4_POST_INIT_CPLT);
 
       nfc_cb.p_nci_init_rsp = p_msg;
-      nfc_cb.p_hal->core_initialized(p_msg->len - p_msg->offset, p);
+      nfc_cb.p_hal->core_initialized(p_msg->len, p);
     }
   } else {
     if (nfc_cb.nci_version == NCI_VERSION_UNKNOWN) {
-- 
2.17.0

